SplitFiles/control.c

```diff
@@ -11,6 +11,56 @@
 // Compile with:
 // zcc +zx -vn -startup=1 -clib=sdcc_iy -D_TEST_CONTROL control.c -o control -create-app
 
+char* buffer_getcommand(const char c) {
+    char* retVal = "          ";
+    switch(c) {
+        case 0:
+            break;
+        case 'B':
+            retVal = "BROWN BREAD";
+            break;
+        case 'W':
+            retVal = "WHITE BREAD";
+            break;
+        case 'T':
+            retVal = "TOAST IT!";
+            break;
+        case 'P':
+            retVal = "POP SLOT";
+            break;
+        case '1':
+        case '2':
+        case '3':
+        case '4':
+        case '5':
+        case '6':
+        case '7':
+        case '8':
+        case '9':
+            sprintf(retVal, "SLOT %c", c);
+            break;
+        default:
+            retVal = "YOU WOT MATE?";
+    }
+    return retVal;
+}
+
+//show the stack on the screen
+void buffer_restack(const ControlBuffer* const buff) {
+    int i;
+    char* command;
+    const int buffer_index = buff->bufferIndex;
+    // first clean the buffer
+    for (i = 2; i <= CONTROL_BUFFER_SIZE; i++) {
+        printf(PRINTAT"%c%c""            \0", 21, i);
+    }
+    // now write out stack
+    for (i = 2; i <= buffer_index; i++) {
```

Comment:
Confused - why start from i=2 but use i-1 as the index?

Reply:
You're supposed to comment before you merge the code!!

This index issue was at the heart of my problems. Your code inserted at 0 at the start but only inserted at 1 thereafter. Trying to make it consistent so it started from 1 at all times.

Reply:
The i=2 was simply a convenience for the PRINTAT though

Reply:
Was hardly going to get snotty over a PR at this stage - not like I "own" the project.

This is the insert (push) code:
    (buff->buffer)[buff->bufferIndex] = c;
    buff->bufferIndex = buff->bufferIndex + 1;
    (buff->buffer)[buff->bufferIndex] = 0;
bufferIndex started out at 0, so the first character went there.  Reason for the third line above was to ensure we always had a string terminator so printing the buffer would stop at the right place (the format added spaces thereafter).